[CI] Failing tests in 1.2.0 branch

https://xgboost-ci.net/blue/organizations/jenkins/xgboost/detail/release_1.2.0/11/pipeline/267
```
>   raise XGBoostError(py_str(_LIB.XGBGetLastError()))
E   xgboost.core.XGBoostError: [03:49:09] ../src/common/device_helpers.cu:64: 
  Check failed: n_uniques == world (1 vs. 4) : Multiple processes within communication group running on
  same CUDA device is not supported
```

This is blocking 1.2.0 release.

Failing tests
* `tests/python-gpu/test_gpu_with_dask.py::TestDistributedGPU::test_dask_array`
* `tests/python-gpu/test_gpu_with_dask.py::TestDistributedGPU::test_dask_dataframe`
* `tests/python-gpu/test_gpu_with_dask.py::TestDistributedGPU::test_empty_dmatrix`
* `tests/python-gpu/test_gpu_with_dask.py::TestDistributedGPU::test_gpu_hist`
* `tests/python-gpu/test_gpu_with_dask.py::test_with_asyncio`

I am able to reproduce the issue.

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -92,7 +92,7 @@ pipeline {
             'test-python-gpu-cuda10.2': { TestPythonGPU(host_cuda_version: '10.2') },
             'test-python-gpu-cuda11.0-cross': { TestPythonGPU(artifact_cuda_version: '10.0', host_cuda_version: '11.0') },
             'test-python-gpu-cuda11.0': { TestPythonGPU(artifact_cuda_version: '11.0', host_cuda_version: '11.0') },
-            'test-python-mgpu-cuda10.2': { TestPythonGPU(artifact_cuda_version: '10.2', host_cuda_version: '10.2', multi_gpu: true) },
+            'test-python-mgpu-cuda10.2': { TestPythonGPU(artifact_cuda_version: '10.0', host_cuda_version: '10.2', multi_gpu: true) },
             'test-cpp-gpu-cuda10.2': { TestCppGPU(artifact_cuda_version: '10.2', host_cuda_version: '10.2') },
             'test-cpp-gpu-cuda11.0': { TestCppGPU(artifact_cuda_version: '11.0', host_cuda_version: '11.0') },
             'test-jvm-jdk8-cuda10.0': { CrossTestJVMwithJDKGPU(artifact_cuda_version: '10.0', host_cuda_version: '10.0') },

--- a/tests/ci_build/Dockerfile.gpu
+++ b/tests/ci_build/Dockerfile.gpu
@@ -17,8 +17,8 @@ ENV PATH=/opt/python/bin:$PATH
 
 # Create new Conda environment with cuDF, Dask, and cuPy
 RUN \
-    conda create -n gpu_test -c rapidsai -c nvidia -c conda-forge -c defaults \
-        python=3.7 cudf=0.14 cudatoolkit=$CUDA_VERSION dask dask-cuda dask-cudf cupy \
+    conda create -n gpu_test -c rapidsai-nightly -c rapidsai -c nvidia -c conda-forge -c defaults \
+        python=3.7 cudf=0.15* cudatoolkit=$CUDA_VERSION dask dask-cuda dask-cudf cupy \
         numpy pytest scipy scikit-learn pandas matplotlib wheel python-kubernetes urllib3 graphviz hypothesis
 
 ENV GOSU_VERSION 1.10

